backgroundfield/backgroundfield.cpp

```diff
@@ -117,13 +117,13 @@ void setBackgroundField(
             //Volume averages
             for(unsigned int fComponent=0;fComponent<3;fComponent++){
                T3DFunction valueFunction = std::bind(bgFunction, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3, (coordinate)fComponent, 0, (coordinate)0);
-               BgBGrid.get(x,y,z)->at(fsgrids::bgbfield::BGBXVOL+fComponent) += volumeAverage(valueFunction,accuracy,start,end);
+               BgBGrid.get(x,y,z)[fsgrids::bgbfield::BGBXVOL+fComponent] += volumeAverage(bgFunction,accuracy,start,end);
```

Comment:
This looks like perhaps a copy-paste error, going from valueFunction to bgFunction? Same for lines 124 and 126, going from derivFunction to bgFunction.